thekuuzzz/PrefParser.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import division
+from __future__ import print_function
+
+import datetime
+import json
+import os
+import sys
+
+
+def main():
+    if not os.path.exists('Preferences'):
+            print("Not a valid path")
+            sys.exit
+    else:
+        with open('Preferences', 'rb') as fh:
+            json_object = json.load(fh)
+            print("Parsing chrome Preferences file")
+            ChromePreferenceParser.parseName(json_object)
+            ChromePreferenceParser.parseDate(json_object)
+            ChromePreferenceParser.parseMedia(json_object)
+            ChromePreferenceParser.parseExt(json_object)
+            ChromePreferenceParser.parsePrint(json_object)
+
+
+class ChromePreferenceParser(object):
+
+    def _findIn(searchSpace, item):
+        # Recursive search for items inside a nested dictionary
+        foundItem = []
+        try:
+            for c, d in searchSpace.items():
+                if c == item:
+                    foundItem.append(d)
+                else:
+                    if isinstance(d, dict):
+                        nestResult = findIn(d, item)
+                        for result in nestResult:
+                            foundItem.append(result)
+                        else:
+                            if isinstance(d, dict):
+                                for nestedItems in d:
+                                    if isinstance(nestedItems, dict):
+                                        extraItems = findIn(nestedItems, item)
+                                        for evenMoreItems in extraItems:
+                                            foundItem.append(extraItems)
+            return foundItem
+        except Exception as e:
+            print("Exception running application, here's what we know")
+            print(e)
+            return 0
+
+    def _dateConversion(webkitTimestamp):
+        # Converts date from webkit time to ISO 8601 time
+        startDate = datetime.datetime(1601, 1, 1, 0, 0, 0, 0)
+        currentDate = datetime.timedelta(microseconds=int(webkitTimestamp))
+        isoTime = startDate + currentDate
+        return isoTime
+
+    def parseName():
+        # Parse through and get the account name, email, and profile name
+        # Returns the name, email and profile name
+        try:
+
+            for d in json_object['account_info']:
+                name = d.get('full_name')
+                email = d.get('email')
+                for c in d['profile']:
+                    profile = c.get('name')
+
+                    return name, email, profile
+        except Exception as e:
+            print('Exception caught: ' + e)
+            raise EnvironmentError.filename
+            sys.exit
+
+    def parseDate():
+        # Parses the last time chrome was updated in ISO 8601 UTC format
+        # Returns the last update date in ISO 8601 UTC
+        try:
+
+            webKitTime = json_object['last_update_date']
```

Comment:
I would also use .get('last_update_date') 

and thus avoid the potential KeyError and need for error handling on this particular one, unless you expect other errors to come up, like you may get from _DateConversion (if the date is improperly formatted) and there you are casting a string to an int... so one error you may get from that code is simple to test....

```
$ ipython
In [1]: int('not a number')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-cd39f7b9e207> in <module>()
----> 1 int('not a number')

ValueError: invalid literal for int() with base 10: 'not a number'

```

You see.. you will get a ValueError from that....